js/apps/events-page/controllers/eventDetailsCtrl.js

```diff
@@ -2,23 +2,35 @@
 	'use strict';
 
 	const EventDetailsCtrl = function EventsPageCtrl($scope, $timeout, $routeParams, CONSTANTS, eventsService, dateUtilityService) {
-		const self = this;
+		const vm = this;
 		const id = $routeParams.id;
 
-		self.data = {};
-		self.data.EventStartDate = '';
-		self.data.EventStartTime = '';
-		self.data.EventEndTime = '';
+		vm.data = {};
+		vm.data.EventStartDate = '';
+		vm.data.EventStartTime = '';
+		vm.data.EventEndTime = '';
+		vm.isLoading = true;
+		vm.isError = false;
+		vm.requestErrorMessage = 'Unfortunately, there was a problem losding this event\'s details. Please try again in a few minutes.';
 
 		const processEventData = (data) => {
-			self.data = data;
-			self.data.EventStartDate = moment(self.data.EventStart).format('MMMM D, YYYY');
-			self.data.EventSchedule = dateUtilityService.formatSchedule(self.data.EventStart, self.data.EventLength, self.data.AllDay);
-			self.isRegistrationRequired = self.data.RegistrationTypeCodeEnum !== 0;
-			self.isOver = moment().isAfter(moment(self.data.EventStart).add(self.data.EventLength, 'm'));
+			vm.data = data;
+			vm.data.EventStartDate = moment(vm.data.EventStart).format('MMMM D, YYYY');
```

Comment:
I'm assuming we will always have a start date?

Reply:
That is correct.